image_captioning/models/decoder.py

```diff
@@ -0,0 +1,120 @@
+"""
+Decoder model.
+"""
+from typing import Tuple
+
+import tensorflow as tf
+
+from .attention import BahdanauAttention
+
+
+class Decoder(tf.keras.Model):
+    """Decoder model.
+    """
+
+    def __init__(self, hidden_units: int, attn_hidden_units: int,
+                 vocab_size: int, embedding_dim: int):
+        """
+        Arguments:
+            hidden_units: Number of hidden units for LSTM.
+            attn_hidden_units: Number of hidden units for attention module.
+            vocab_size: Number of possible vocabularies in the language.
+            embedding_dim: Dimension of word embedding vector.
```

Comment:
I think we might want to use `word_embedding_dim` instead of `embedding_dim` to alleviate ambiguation between image embedding dimensions and word embedding dimensions